packages/components/src/Listing.tsx

```diff
@@ -0,0 +1,85 @@
+import MonacoEditor, { useMonaco } from "@monaco-editor/react";
+import { compileDomain } from "@penrose/core";
+import { editor, IRange } from "monaco-editor";
+import { useCallback, useEffect } from "react";
+import {
+  SubstanceCompletions,
+  SubstanceConfig,
+  SubstanceLanguageTokens,
+} from "./editing/languages/SubstanceConfig";
+
+export const monacoOptions: editor.IStandaloneEditorConstructionOptions = {
+  automaticLayout: true,
+  readOnly: true,
+  minimap: { enabled: false },
+  wordWrap: "on",
+  lineNumbers: "off",
+  fontSize: 16,
+  scrollbar: {
+    handleMouseWheel: false,
+  },
+  scrollBeyondLastLine: false,
+  renderLineHighlight: "none",
+};
+
+const Listing = ({
+  domain,
+  substance,
+  width,
+  height,
+}: {
+  domain: string;
+  substance: string;
+  width: string;
+  height: string;
+}) => {
+  const env = compileDomain(domain).unsafelyUnwrap();
```

Comment:
Not in scope: same thing as the diagram component, some error handling would be good.